pallets/capacity/src/migration/v2.rs

```diff
@@ -0,0 +1,112 @@
+use crate::{
+	types::{StakingDetails, UnlockChunk},
+	BalanceOf, Config, Pallet, StakingAccountLedger, StakingType, UnlockChunks, UnstakeUnlocks,
+};
+use frame_support::{
+	pallet_prelude::{GetStorageVersion, Weight},
+	traits::{OnRuntimeUpgrade, StorageVersion},
+	weights::constants::RocksDbWeight,
+};
+const LOG_TARGET: &str = "runtime::capacity";
+
+#[cfg(feature = "try-runtime")]
+use sp_std::{fmt::Debug, vec::Vec};
+
+/// Only contains V1 storage format
+pub mod v1 {
+	use super::*;
+	use codec::{Decode, Encode, MaxEncodedLen};
+	use frame_support::{storage_alias, BoundedVec, Twox64Concat};
+	use scale_info::TypeInfo;
+
+	#[derive(Default, Encode, Decode, PartialEq, Debug, TypeInfo, Eq, MaxEncodedLen)]
+	/// Old StakingAccountDetails struct
+	pub struct StakingAccountDetails<T: Config> {
+		/// The amount a Staker has staked, minus the sum of all tokens in `unlocking`.
+		pub active: BalanceOf<T>,
+		/// The total amount of tokens in `active` and `unlocking`
+		pub total: BalanceOf<T>,
+		/// Unstaked balances that are thawing or awaiting withdrawal.
+		pub unlocking: BoundedVec<
+			UnlockChunk<BalanceOf<T>, T::EpochNumber>,
+			<T as crate::pallet::Config>::MaxUnlockingChunks,
+		>,
+	}
+
+	#[storage_alias]
+	/// alias to StakingAccountLedger storage
+	pub(crate) type StakingAccountLedger<T: Config> = StorageMap<
+		Pallet<T>,
+		Twox64Concat,
+		<T as frame_system::Config>::AccountId,
+		StakingAccountDetails<T>,
+	>;
+}
+
+/// migrate StakingAccountLedger to use new StakingDetails
+pub fn migrate_to_v2<T: Config>() -> Weight {
+	let on_chain_version = Pallet::<T>::on_chain_storage_version();
+
+	if on_chain_version.lt(&2) {
+		log::info!(target: LOG_TARGET, "🔄 StakingAccountLedger migration started");
+		let mut maybe_count = 0u32;
+		StakingAccountLedger::<T>::translate(
+			|key: <T as frame_system::Config>::AccountId,
+			 old_details: v1::StakingAccountDetails<T>| {
+				let new_account: StakingDetails<T> = StakingDetails {
+					active: old_details.active,
+					staking_type: StakingType::MaximumCapacity,
+				};
+				let new_unlocks: UnlockChunks<T> =
+					UnlockChunks { unlocking: old_details.unlocking };
+				UnstakeUnlocks::<T>::insert(key, new_unlocks);
+				maybe_count += 1;
+				log::info!(target: LOG_TARGET,"migrated {:?}", maybe_count);
+				Some(new_account)
+			},
+		);
+		StorageVersion::new(2).put::<Pallet<T>>();
+		log::info!(target: LOG_TARGET, "🔄 migration finished");
+		let count = (StakingAccountLedger::<T>::iter().count() + 1) as u64;
+		RocksDbWeight::get().reads_writes(count, count)
```

Comment:
it would be great to print weight so that we know it would fit in a block

Reply:
Rococo (I reformatted the weight):
```shell
...
[2023-11-15T00:35:34Z INFO  runtime::capacity] migrated 336
[2023-11-15T00:35:34Z INFO  runtime::capacity] 🔄 migration finished
[2023-11-15T00:35:34Z INFO  runtime::capacity] Migration calculated weight = Weight { ref_time: 75_725_000_000, proof_size: 0 }
```

Mainnet:
```shell
[2023-11-15T00:55:08Z INFO  runtime::capacity] 🔄 StakingAccountLedger migration started
[2023-11-15T00:55:08Z INFO  runtime::capacity] migrated 1
[2023-11-15T00:55:08Z INFO  runtime::capacity] migrated 2
[2023-11-15T00:55:08Z INFO  runtime::capacity] 🔄 migration finished
[2023-11-15T00:55:08Z INFO  runtime::capacity] Migration calculated weight = Weight { ref_time: 575000000, proof_size: 0 }
[2023-11-15T00:55:08Z INFO  runtime::capacity] ✅ migration post_upgrade checks passed
```